src/model_signing/_cli.py

```diff
@@ -0,0 +1,451 @@
+    Alternatively, a constant identity token can be provided via
+    `--identity_token`.
+
+    Sigstore allows users to use a staging instance for test-only signatures.
+    Passing the `--use_staging` flag would use that instance instead of the
+    production one.
+    """
+    signer = sigstore.SigstoreDSSESigner(
+        use_ambient_credentials=use_ambient_credentials,
+        use_staging=use_staging,
+        identity_token=identity_token,
+    )
+    _serialize_and_sign(model_path, signer, signature)
+
+
+@_sign.command(name="key")
+@_model_path_argument
+@_write_signature_option
+@_private_key_option
+def _sign_private_key(
+    model_path: pathlib.Path, signature: pathlib.Path, private_key: pathlib.Path
+) -> None:
+    """Sign using a private key (paired with a public one).
+
+    Signing the model at MODEL_PATH, produces the signature at SIGNATURE_PATH
+    (as per `--signature` option).
+
+    Traditionally, signing could be achieved by using a public/private key pair.
+    Pass the signing key using `--private_key`.
+
+    Note that this method does not provide a way to tie to the identity of the
+    signer, outside of pairing the keys. Also note that we don't offer key
+    management protocols.
+    """
+    signer = in_toto_signature.IntotoSigner(
+        # TODO: The API needs clean up, it uses str, not pathlib.Path
+        key.ECKeySigner.from_path(private_key_path=private_key.as_posix())
+    )
+    _serialize_and_sign(model_path, signer, signature)
+
+
+@_sign.command(name="certificate")
+@_model_path_argument
+@_write_signature_option
+@_private_key_option
+@click.option(
+    "--signing_certificate",
+    type=pathlib.Path,
+    metavar="CERTIFICATE_PATH",
+    required=True,
+    help="Path to the signing certificate, as a PEM-encoded file.",
+)
+@_certificate_root_of_trust_option
+def _sign_certificate(
+    model_path: pathlib.Path,
+    signature: pathlib.Path,
+    private_key: pathlib.Path,
+    signing_certificate: pathlib.Path,
+    certificate_chain: list[pathlib.Path],
+) -> None:
+    """Sign using a certificate.
+
+    Signing the model at MODEL_PATH, produces the signature at SIGNATURE_PATH
+    (as per `--signature` option).
+
+    Traditionally, signing can be achieved by using keys from a certificate.
+    The certificate can also provide the identity of the signer, making this
+    method more informative than just using a public/private key pair for
+    signing.  Pass the private signing key using `--private_key` and signing
+    certificate via `--signing_certificate`. Optionally, pass a certificate
+    chain via `--certificate_chain` to establish root of trust (this option can
+    be repeated as needed, or all cerificates could be placed in a single file).
+
+    Note that we don't offer certificate and key management protocols.
+    """
+    if not certificate_chain:
+        raise click.BadOptionUsage(
+            "--certificate_chain", "No certificate chain was provided"
+        )
```

Comment:
wouldn't this conflict with `--certificate_chain` being optional?

Reply:
Oh, this was debug stuff that I forgot to remove. Great catch!